tests/codebuild/create_tests.sh

```diff
@@ -80,15 +80,26 @@ function verify_debug_test
 {
   local crd_type="$1"
   local crd_instance="$2"
-  local expected_debug_job_status="$3"
+  local timeout="$3"
+  local expected_debug_job_status="$4"
+  # First verify that trainingjob has been completed
+  verify_test TrainingJob xgboost-mnist-debugger $timeout Completed
+
   # TODO extend this for multiple debug job with debug job statuses parameter
-  local found_debug_job_status="$(kubectl get "$crd_type" "$crd_instance" -o json | jq -r '.status.debugRuleEvaluationStatuses[0].ruleEvaluationStatus')"
-  if [ "${expected_debug_job_status}" == "${found_debug_job_status}" ]; then
-      echo "[PASSED] Verified ${crd_type} ${crd_instance} debug job has status ${expected_debug_job_status}"
-  else
-      echo "[FAILED] ${crd_type} ${crd_instance} debug job has status ${found_debug_job_status} while expected status was ${expected_debug_job_status}"
-      exit 1
+
+  echo "Waiting for debug job to finish"
+  timeout "${timeout}" bash -c \
+    'until [ "$(kubectl get "$0" "$1" -o json | jq -r .status.debugRuleEvaluationStatuses[0].ruleEvaluationStatus)" != "$2" ]; do \
+      echo "Debug job has not completed yet"; \
+      sleep 1; \
+    done' "${crd_type}" "${crd_instance}" "${expected_debug_job_status}"
```

Comment:
Is this logic inverted? I don't think this will wait until the zeroth rule evaluation status is `NoIssuesFound`; it will wait until the zeroth rule evaluation status _is not_ `NoIssuesFound`.

Look at this test code I made:

```bash
#!/usr/bin/env bash

timeout=5s
crd_type=CRD_TYPE
crd_instance=CRD_INSTANCE
actual_debug_job_status="$1"
expected_debug_job_status="NoIssuesFound"


set -x
timeout "${timeout}" bash -c \
  'set -x; until [ "$3" != "$2" ]; do \
    echo "Debug job has not completed yet"; \
    sleep 1; \
  done' "${crd_type}" "${crd_instance}" "${expected_debug_job_status}" "${actual_debug_job_status}"

#timeout "${timeout}" bash -c \
#  'until [ "$(kubectl get "$0" "$1" -o json | jq -r .status.debugRuleEvaluationStatuses[0].ruleEvaluationStatus)" != "$2" ]; do \
#    echo "Debug job has not completed yet"; \
#    sleep 1; \
#  done' "${crd_type}" "${crd_instance}" "${expected_debug_job_status}"
```

Testing it:
```bash
$ $ ./script NoIssuesFound
+ timeout 5s bash -c 'set -x; until [ "$3" != "$2" ]; do \
    echo "Debug job has not completed yet"; \
    sleep 1; \
  done' CRD_TYPE CRD_INSTANCE NoIssuesFound NoIssuesFound
+ '[' NoIssuesFound '!=' NoIssuesFound ']'
+ echo 'Debug job has not completed yet'
Debug job has not completed yet
+ sleep 1
+ '[' NoIssuesFound '!=' NoIssuesFound ']'
+ echo 'Debug job has not completed yet'
Debug job has not completed yet
+ sleep 1
+ '[' NoIssuesFound '!=' NoIssuesFound ']'
+ echo 'Debug job has not completed yet'
Debug job has not completed yet
+ sleep 1
+ '[' NoIssuesFound '!=' NoIssuesFound ']'
+ echo 'Debug job has not completed yet'
Debug job has not completed yet
+ sleep 1
+ '[' NoIssuesFound '!=' NoIssuesFound ']'
+ echo 'Debug job has not completed yet'
Debug job has not completed yet
+ sleep 1
```

vs.

```bash
$ ./script InProgress
+ timeout 5s bash -c 'set -x; until [ "$3" != "$2" ]; do \
    echo "Debug job has not completed yet"; \
    sleep 1; \
  done' CRD_TYPE CRD_INSTANCE NoIssuesFound InProgress
+ '[' InProgress '!=' NoIssuesFound ']'
```

This code here waits until the status is _not_ `NoIssuesFound`; I think you want the opposite.